### ISSUE TITLE
PREG_BACKTRACK_LIMIT_ERROR $search0 optimize

### ISSUE THREAD

**Issue by thundor** [vanyog/moodle-filter_multilangsecond#4]
I had a heavily nested html enclosed by <div lang=*> tags and the regex callback for $search0 returned null because of PREG_BACKTRACK_LIMIT_ERROR. You may use my fix
`$search0 = '/(?:<([a-z]+)[^>]*?lang=".*?".*?>.*?<\/\1>).*<\1[^>]*?lang=".*?".*?>.*?<\/\1>/is';`
Edit: THIS IS NOT CORRECT!

**Comment by vanyog** [comment]
This will not work with heading tags  h1, h2, and so on. And will remove everything between first and last tags with lang attributes. In case like this:
&lt;p lang="bg"&gt;Заглавие&lt;/p&gt;
<br>&lt;p&gt;aaa&lt;/p&gt;
<br>&lt;p lang="en"&gt;Heading&lt;/p&gt;
the filter must do nothing, but with your change it will leave only the first or the last paragraph depending on the current language.

**Comment by thundor** [comment]
True, thank you, i will continue to attempt to optimize it as I need to do so urgently
